dev-dotnet/dotnet-sdk/dotnet-sdk-5.0.203.ebuild

```
@@ -0,0 +1,238 @@
+	#  but the reference-packages have just been installed in /usr/lib
+	for x in "${DOTNET_ROOT}"/reference-packages "${BROOT}"/usr/lib/dotnet/reference-packages; do
+		if [[ -d "${x}" ]]; then
+			DOTNET_REFERENCE="${x}"
+			break
+		fi
+	done
+
+	if [[ ! -d "${DOTNET_REFERENCE}" ]]; then
+		die "Can't find installed .NET Source-build Reference Packages"
+	fi
+}
+
+src_unpack() {
+	unpack "${MY_P}.tar.gz"
+	dotnet_unpack
+}
+
+src_prepare() {
+	# add_patch <submodule> <file.patch>
+	add_patch() {
+		cp "${2}" "patches/${1}/9000-$(basename "${2}")" || die
+	}
+
+	eapply "${FILESDIR}"/dotnet-sdk-5.0.203-additional-arguments.patch
+	add_patch sdk "${DISTDIR}"/dotnet-sdk-telemetry-optout.patch
+	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-add-gentoo-ids.patch
+	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-additional-arguments.patch
+	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-as-needed.patch
+	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-locate-ranlib.patch
+
+	default
+}
+
+src_configure() {
+	dotnet_find
+
+	# SDK is modified in-place...
+	if use system-bootstrap; then
+		mkdir -p dotnet || die
+		ln -s "${DOTNET_ROOT}"/{host,packs,shared} dotnet || die
+		cp -a "${DOTNET_ROOT}"/{dotnet,sdk} dotnet || die
+	else
+		ln -s "${DOTNET_ROOT}" dotnet || die
+	fi
+}
+
+src_compile() {
+	dotnet_find
+	dotnet_find_reference
+
+	# Required for --with-packages to work
+	rm -rf packages/archive
+
+	local mybuildargs=(
+		--with-sdk dotnet
+		--with-packages "${DOTNET_ROOT}"/source-artifacts
+		--with-ref-packages "${DOTNET_REFERENCE}"
+		--
+
+		/p:UseSystemLibraries=true
+		/p:UseSystemLibunwind=$(usex system-libunwind true false)
+
+		/p:LogVerbosity=normal
+		/p:MinimalConsoleLogOutput=false
+		/verbosity:normal
+
+		# Not sure if this has any effect on the subprocesses
+		/maxCpuCount:$(makeopts_jobs)
+
+		# /p:AdditionalBuildArguments are passed to src/runtime.*/eng/build.sh
+		# /p:NativeBuildArguments are passed to src/runtime.*/eng/native/build-commons.sh
+		/p:AdditionalBuildArguments="-cmakeargs -DCMAKE_VERBOSE_MAKEFILE=ON"
+		/p:NativeBuildArguments="${CMAKE_GENERATOR_FLAG} -numproc $(makeopts_jobs)"
+		# If any dotnet maintainer is reading this: Please honor the
+		#  CMAKE_MAKEFILE_GENERATOR and MAKEFLAGS variables if set.
+	)
+
+	# Adapted from www-client/firefox
+	if ${DOTNET_FORCE_CLANG} && ! tc-is-clang; then
+		einfo "Enforcing the use of Clang due to build errors..."
+		AR=llvm-ar
+		CC=${CHOST}-clang
+		CXX=${CHOST}-clang++
+		NM=llvm-nm
+		OBJCOPY=llvm-objcopy
+		OBJDUMP=llvm-objdump
+		RANLIB=llvm-ranlib
+
+		if is-flagq '-flto*' && ! has_version 'sys-devel/clang[default-lld]' && ! is-ldflagq '-fuse-ld=lld'; then
+			has_version sys-devel/lld || die "Please install sys-devel/lld (or sys-devel/clang[default-lld]) to build with LTO"
+			einfo "Enforcing the use of LLD for Clang LTO..."
+			append-ldflags '-fuse-ld=lld'
+		fi
+
+		strip-unsupported-flags
```

```suggestion
		fi

		has_version 'sys-devel/clang[default-libcxx]' && append-cxxflags -stdlib=libstdc++
		strip-unsupported-flags
```

Build fails with -stdlib=libcxx